File: CSC330/Homework/perl/recursion/recursion.pl
#!/usr/bin/perl

sub fact {
 my $val = $_[0];
 if ( $val < 0 )
 {
   return "-- ERROR --\n";
 }
 elsif ( $val == 0 || $val == 1) {
    return 1;
 }
 else {
    return $val * fact($val-1);
   }
}

# Take the command line input, compute the factorial of it, and print
# it out.
print fact($ARGV[0]) . "\n";

File: CSC330/Homework/Python/recursion/recursion.py
# Recursive Method to Compute Factorial
import sys

def fact(val) :
      if(val < 0) :
            return "--NO--\n"
      elif(val == 0) or (val == 1) :
            return 1
      else : 
            return val * fact(val - 1)
            
# Take command line input, compute the factorial, print it out            
userInput = int(sys.argv[1])
print("Factorial of " + str(userInput) + " : " + str(fact(userInput)))